include/nbl/asset/IPipeline.h

```diff
@@ -105,11 +105,14 @@ class IPipeline
 		};
 
 		inline const PipelineLayout* getLayout() const {return m_layout.get();}
+		inline const asset::E_PIPELINE_BIND_POINT getBindPoint() const { return m_bindPoint; }
 
 	protected:
-		inline IPipeline(core::smart_refctd_ptr<const PipelineLayout>&& _layout) : m_layout(std::move(_layout)) {}
+		inline IPipeline(core::smart_refctd_ptr<const PipelineLayout>&& _layout, asset::E_PIPELINE_BIND_POINT bindPoint)
+      : m_layout(std::move(_layout)), m_bindPoint(bindPoint) {}
 
 		core::smart_refctd_ptr<const PipelineLayout> m_layout;
+		asset::E_PIPELINE_BIND_POINT m_bindPoint;
```

Comment:
the proper way to do it is to add
```cpp
virtual const  E_PIPELINE_BIND_POINT getBindPoint() const = 0;
```
and then override in derived classes.

Passing a thing from a constructor and then setting it as a member is bug/error prone, bad practice and eats up memory in each instance of the base class (whereas a virtual override returning a constant literal consumes zero extra mem)

Thats the pattern we use for things like `IAsset::getType` amongst others.